CONTRIBUTING.md

```diff
@@ -82,4 +82,8 @@ All SR-IOV tests must be identified via either a tag or a commit hash, represent
 reference specific tests. For this reason, the "common name" a test receives is not considered its formal reference, and may
 remain even if a test changes. Semantic versioning of tags should be observed to create relevant links to unique tests. For more information, see the root level `README.md`, [Releases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository), and [Semantic Versioning](https://semver.org/).
 
-As this identifier is used to identify a specific, unique test definition, a change to the tag is required whenever the functionality/content of a test case changes, inclusive of bug fixes. Practically, this means that when changes are merged to main the tag and release should change. Maintaining proper versioning should be a main consideration of the repository maintainers after accepting a pull request and merging to main.
+As this identifier is used to identify a specific, unique test definition, a change to the tag is required whenever the functionality/content of a test case changes, inclusive of bug fixes. Practically, this means that when changes are merged to main the tag and release should change. Maintaining proper versioning should be a main consideration of the repository maintainers after accepting a Pull Request and merging to main.
+
+## Validating Changes
+
+All SR-IOV and common tests should be validated to work and, in cases where changes affect common code, that changes do not break existing tests or common code. To this end there are GitHub Actions running to validate formatting by way of a linter, a series of mock tests checking the common code, as well as an option to run tests on a lab machine. The linter and mocks run automatically on Pull Requests. The optional "real" testing can be run by applying the "e2e-test" label to the Pull Request. This will start an action which queues either an end-to-end test (where common code is affected) or runs the specific test modified (when no common code is affected). Because these tests take both time and resources, they should only be started after the PR is ready to merge as a final check of functionality. After the test run, the Action will remove the "e2e-test" label.
```

Comment:
Should we mention here who is supposed/allowed to use e2e-test label? Formatting-wise, I suggest a new  paragraph starting at "The optional "real" testing...

Reply:
@akiselev1 Addressed in the most recent commit